crates/core/src/factories/core_space.rs

```diff
@@ -198,6 +198,39 @@ impl TxSpaceHandler for TxHandlerTranslator {
     ) -> K2Result<()> {
         self.0.recv_notify(peer, space, data)
     }
+
+    fn mark_peer_unresponsive(&self, peer: Url) -> K2Result<()> {
+        let core_space = self.1.upgrade().ok_or(K2Error::Other {
+            ctx: "CoreSpace had been dropped.".into(),
+            src: DynInnerError(None),
+        })?;
+        tokio::task::spawn(async move {
+            // Only add a peer as unreachable to the peer meta store if it is
+            // also in the peer store. That's because this method may be called
+            // from a context that has no awareness about which space a peer
+            // (Url) is part of and that therefore needs to iteratively call
+            // it for all spaces
+            let peers = core_space.peer_store.get_all().await?;
+            match peers.iter().find(|p| p.url == Some(peer.clone())) {
+                Some(agent_info) => {
+                    if let Err(e) = core_space
+                        .peer_meta_store
+                        .mark_peer_unresponsive(
+                            agent_info.url.clone().unwrap(),
+                            agent_info.expires_at,
+                        )
+                        .await
+                    {
+                        tracing::error!("Failed to mark peer at url {:?} as unresponsive in the peer meta store: {e}", agent_info.url);
```

Comment:
We use an inconsistent style for this through K2 but I'm starting to prefer the way that this is rendered if you use:

`tracing::error!(?err, "some message with more context {}", agent_info.url);`

Entirely a style preference for how the output looks and not a "must do" comment :)

Reply:
3efccaaf4434f6b38331ecd477ae48f224553f6d